### PR TITLE
adding a tiny configuration library

### PR DESCRIPTION
**Pull request by theocat**
- adding a tiny configuration library for more convenient access to the configuration
 - adding refined types

### CODE REVIEW

File: config/build.sbt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import pulse.plugin._
+import pulse.plugin.dependencies._
+import local.dependencies._
+
+libraryDependencies ++= Seq(
+  typesafe.config,
+  "co.fs2" %% "fs2-cats" % "0.3.0",
```
Comment (theocat) [inline config/build.sbt]: @eugene-cheverda can you please move this interop dependency and refined to `sbt-pulse` ?

File: config/src/main/scala/pulse/config/TypeDescriptor.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+trait TypeDescriptor[T] {
+
+  val value: DescriptorValue
+
+}
+
+trait DescriptorValue extends Product with Serializable
+
+object Values {
+  case object String  extends DescriptorValue
```
Comment (theocat) [inline config/src/main/scala/pulse/config/TypeDescriptor.scala]: @vbachurin can you please work on supporting more types?

File: project/local.scala [hunk 3]
```diff
@@ -7,12 +7,22 @@ object local {
...
+      val config  = "1.3.1"
+      val refined = "0.8.0"
     }
 
     object log4s {
       val core = "org.log4s" %% "log4s" % versions.log4s
     }
+
+    object typesafe {
+      val config = "com.typesafe" % "config" % versions.config
```
Comment (theocat) [inline project/local.scala]: @eugene-cheverda should be moved to `sbt-pulse`

File: project/local.scala [hunk 4]
```diff
@@ -7,12 +7,22 @@ object local {
...
+    }
+
+    object refined {
+      val core = "eu.timepit" %% "refined" % versions.refined
```
Comment (theocat) [inline project/local.scala]: should be moved to `sbt-pulse`

File: config/build.sbt (hunk 1 above)
Comment (eugene-cheverda) [inline config/build.sbt]: done

File: project/local.scala (hunk 3 above)
Comment (eugene-cheverda) [inline project/local.scala]: done

File: project/local.scala (hunk 4 above)
Comment (eugene-cheverda) [inline project/local.scala]: done

File: config/src/main/scala/pulse/config/TypeDescriptor.scala (hunk 2 above)
Comment (vbachurin) [inline config/src/main/scala/pulse/config/TypeDescriptor.scala]: @theocat please name a few of the highest priority

**Pull request merged**